### PR TITLE
appsettings updated

### LINKED ISSUE #230: Refactor testclientoptions in appsettings.json
This change involves updating the TestClientOptions sestion in SkillsFunctionalTests/appSettings.json to be able to store the bot appId in config rather than in code. 

It also involves changing the structure of appsettings so the directline client options can be loaded in a dictionary with well known bot names rather than in an array.

### PR DESCRIPTION
**Pull request by gabog**
Fixes #230

- Modified structure of appsettings.json so it can be loaded in a dictiorany and we don't have the BotIDs in code.
- Updated uses of TestOptions to consume the new changes.
- Updated yaml files to 
- Renamed some bots and settings to use DotNet instead of dotnet

### CODE REVIEW

File: Libraries/TranscriptTestRunner/TestClients/DirectLinetTestClientOptions.cs [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-}
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace TranscriptTestRunner.TestClients
+{
+    /// <summary>
+    /// Class with the options needed to run a test against a TesClient.
+    /// </summary>
+    public class DirectLinetTestClientOptions
```
Comment (ceciliaavila) [inline Libraries/TranscriptTestRunner/TestClients/DirectLinetTestClientOptions.cs:9]:
```suggestion
    public class DirectLineTestClientOptions
```

File: Tests/SkillFunctionalTests/Common/HostBot.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// Licensed under the MIT License.
+
+namespace SkillFunctionalTests.Common
+{
+    public enum HostBot
+    {
+        /// <summary>
+        /// Simple host implemented using dotnet 3.1.
+        /// </summary>
+        SimpleHostBotDotNet,
```
Comment (ceciliaavila) [inline Tests/SkillFunctionalTests/Common/HostBot.cs:11]: Could we rename the bots here to match the names used in the pipelines?
```suggestion
        SimpleHostBotDotnet,
```

File: Tests/SkillFunctionalTests/Common/HostBot.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        /// <summary>
+        /// Simple host implemented using dotnet 2.1.
+        /// </summary>
+        SimpleHostBotDotNet21,
```
Comment (ceciliaavila) [inline Tests/SkillFunctionalTests/Common/HostBot.cs:16]:
```suggestion
        SimpleHostBot21Dotnet,
```

File: Tests/SkillFunctionalTests/Common/HostBot.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+        /// <summary>
+        /// Simple host implemented using Python.
+        /// </summary>
+        SimpleHostBotPython,
+
+        /// <summary>
+        /// Simple host implemented using Composer and the dotnet runtime.
+        /// </summary>
+        SimpleComposerHostBotDotNet,
```
Comment (ceciliaavila) [inline Tests/SkillFunctionalTests/Common/HostBot.cs:31]:
```suggestion
        SimpleComposerHostBotDotnet,
```

File: Tests/SkillFunctionalTests/Common/HostBot.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+        /// <summary>
+        /// Host implemented using dotnet and waterfall dialogs.
+        /// </summary>
+        WaterfallHostBotDotNet,
```
Comment (ceciliaavila) [inline Tests/SkillFunctionalTests/Common/HostBot.cs:36]:
```suggestion
        WaterfallHostBotDotnet,
```

File: Tests/SkillFunctionalTests/Common/HostBot.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+        /// <summary>
+        /// Host implemented using Python and waterfall dialogs.
+        /// </summary>
+        WaterfallHostBotPython,
+
+        /// <summary>
+        /// Host implemented using Composer and the dotnet runtime.
+        /// </summary>
+        ComposerHostBotDotNet
```
Comment (ceciliaavila) [inline Tests/SkillFunctionalTests/Common/HostBot.cs:51]:
```suggestion
        ComposerHostBotDotnet
```

**Comment by gabog** [comment]
Hey @ceciliaavila, is the suggestion because is the correct way? or because it will break something else?

I don't have a strong opinion but from looking at existing nuget packages published by microsoft, we seem to use DotNet and not Dotnet.

image


image


image

**Comment by ceciliaavila** [comment]
> Hey @ceciliaavila, is the suggestion because is the correct way? or because it will break something else?
> 
> I don't have a strong opinion but from looking at existing nuget packages published by microsoft, we seem to use DotNet and not Dotnet.

Ok, no problem. We'll need to change them in the pipelines' yamls to match these, but it shouldn't be a problem.

File: Tests/SkillFunctionalTests/Common/HostBot.cs (hunk 2 above)
Comment (gabog) [reply Tests/SkillFunctionalTests/Common/HostBot.cs:11]: We are standardizing on DotNet (captial N), closing

File: Tests/SkillFunctionalTests/Common/HostBot.cs (hunk 3 above)
Comment (gabog) [reply Tests/SkillFunctionalTests/Common/HostBot.cs:16]: We are standardizing on DotNet (captial N), closing

File: Tests/SkillFunctionalTests/Common/HostBot.cs (hunk 4 above)
Comment (gabog) [reply Tests/SkillFunctionalTests/Common/HostBot.cs:31]: We are standardizing on DotNet (captial N), closing

File: Tests/SkillFunctionalTests/Common/HostBot.cs (hunk 5 above)
Comment (gabog) [reply Tests/SkillFunctionalTests/Common/HostBot.cs:36]: We are standardizing on DotNet (captial N), closing

File: Tests/SkillFunctionalTests/Common/HostBot.cs (hunk 6 above)
Comment (gabog) [reply Tests/SkillFunctionalTests/Common/HostBot.cs:51]: We are standardizing on DotNet (capital N), closing

File: Libraries/TranscriptTestRunner/TestClients/DirectLinetTestClientOptions.cs (hunk 1 above)
Comment (gabog) [reply Libraries/TranscriptTestRunner/TestClients/DirectLinetTestClientOptions.cs]: Done, thanks!

**Comment by gabog** [comment]
@ceciliaavila, @denscollo, this PR is ready for review, take a look and let me know if we can merge it to staging.

Thanks

**Review by denscollo — APPROVED** [review]

**Pull request merged**